GoT/app/controllers/characters_controller.rb

```diff
@@ -0,0 +1,57 @@
+class CharactersController < ApplicationController
+
+
+  # index
+def index
+  @character = Character.all
+end
+
+# new
+def new
+  @house = House.find(params[:house_id])
+  @character = Character.new
+end
+
+# show
+def show
+@character = Character.find(params[:house_id])
+end
+
+
+   def create
+     @house = House.find(params[:house_id])
+     @character = @house.characters.create(character_params)
+     redirect_to "/houses/#{@house.id}"
+   end
+
+
+   def edit
+      @house = House.find(params[:house_id])
+      @character = Character.find(params[:id])
+   end
+
+   # update
+     def update
+       @house = House.find(params[:house_id])
+       @character = @house.characters.find(params[:id])
+        @character.update(character_params)
+
+       redirect_to house_path(@house)
+     end
+
+
+   def destroy
+      @house = House.find(params[:house_id])
+      @character = Character.find(params[:id])
+      @character.destroy
+
+    redirect_to "/houses"
+
+      end
+
+    private
+   def character_params
+     params.require(:character).permit(:name, :photo_url, :title, :quotes, :house_id)
```

Comment:
If you're pulling the `house_id` from the URL, does that mean you also need it inside of your strong params?